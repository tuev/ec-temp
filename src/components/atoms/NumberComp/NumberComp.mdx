import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './NumberComp.stories.tsx'
import NumberComp from './NumberComp.tsx'

<Story name="basic" />
