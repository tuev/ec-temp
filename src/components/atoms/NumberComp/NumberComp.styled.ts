import { space } from 'styled-system'
import { switchProp } from 'styled-tools'
import { COLOR } from 'theme/colors'
import styled, { css } from 'styled-components'

export const AppNumberComp: React.FC = styled.button<{
  color?: 'primary' | 'white'
}>`
  ${space}
  width: 27px;
  height: 27px;
  border: 1px solid ${COLOR.primary};
  border-radius: 50%;
  font-size: 14px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  background-color: transparent;
  outline-color: transparent;
  outline-style: none;
  cursor: pointer;

  ${switchProp(
    'color',
    {
      primary: css`
        background-color: ${COLOR.primary};
        color: ${COLOR.white};
      `,
      white: css`
        background-color: ${COLOR.white};
        color: ${COLOR.primary};
      `,
    },
    css`
      background-color: ${COLOR.primary};
      color: ${COLOR.white};
    `
  )}
`
