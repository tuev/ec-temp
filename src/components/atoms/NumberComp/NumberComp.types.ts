import { SpaceProps } from 'styled-system'
export interface INumberCompProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  data?: number
  color?: 'primary' | 'white'
  children?: null
}

export type BaseNumberProps = INumberCompProps | SpaceProps

export type NumberButton = Omit<INumberCompProps, 'data'>
