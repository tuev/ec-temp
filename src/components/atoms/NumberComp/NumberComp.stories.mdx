import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './NumberComp.stories.tsx'
import NumberComp from './NumberComp.tsx'

<Meta title="NumberComp" component={NumberComp} />
