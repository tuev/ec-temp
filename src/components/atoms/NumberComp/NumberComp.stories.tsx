import React from 'react'
import { action } from '@storybook/addon-actions'
import NumberComp from './NumberComp'

export default {
  component: NumberComp,
  includeStories: ['basic'],
  title: 'AppNumberComp',
}

export const basic = () => (
  <>
    <NumberComp onChange={action('NumberComp Clicked!!!')} data={1} />
    <NumberComp
      onChange={action('NumberComp Clicked!!!')}
      data={22}
      color="white"
    />
    <NumberComp
      onChange={action('NumberComp Clicked!!!')}
      data={102}
      color="white"
    />
  </>
)
