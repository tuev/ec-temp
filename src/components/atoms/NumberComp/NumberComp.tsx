import React, { FC, useMemo } from 'react'
import { AppNumberComp } from './NumberComp.styled'
import { BaseNumberProps } from './NumberComp.types'

const getDataNumber: (data: number) => number = (data) =>
  data < 0 ? 0 : data > 99 ? 99 : data

const NumberComp: FC<BaseNumberProps> = (props: BaseNumberProps) => {
  const { data = 0, ...rest } = props
  const dataNumber: number = useMemo(() => getDataNumber(data), [data])
  return (
    <AppNumberComp color="primary" {...rest}>
      {dataNumber}
    </AppNumberComp>
  )
}
export default NumberComp
