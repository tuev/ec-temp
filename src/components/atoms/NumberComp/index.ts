import NumberComp from './NumberComp'

export { NumberComp }

export default NumberComp
