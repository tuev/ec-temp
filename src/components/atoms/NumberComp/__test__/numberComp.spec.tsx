import React from 'react'
import { mount } from 'enzyme'
import NumberComp from '..'

describe('App number component input', () => {
  it('nmber input component', () => {
    const AppComp = mount(<NumberComp />)

    expect(AppComp).toMatchSnapshot()

    const expectStyle: { [key: string]: string } = {
      width: '27px',
      height: '27px',
      border: '1px solid #F62F5E',
      'border-radius': '50%',
      'font-size': '14px',
      display: 'inline-flex',
      'justify-content': 'center',
      'align-items': 'center',
      'background-color': 'transparent',
      'outline-color': 'transparent',
      'outline-style': 'none',
      cursor: 'pointer',
      'background-color': '#F62F5E',
      color: '#fff',
    }
    for (const key in expectStyle) {
      if (expectStyle[key]) {
        expect(AppComp).toHaveStyleRule(key, expectStyle[key])
      }
    }
  })
})
