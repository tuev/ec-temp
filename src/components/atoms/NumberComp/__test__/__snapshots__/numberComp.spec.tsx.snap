// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App number component input nmber input component 1`] = `
.c0 {
  width: 27px;
  height: 27px;
  border: 1px solid #F62F5E;
  border-radius: 50%;
  font-size: 14px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  outline-color: transparent;
  outline-style: none;
  cursor: pointer;
  background-color: #F62F5E;
  color: #fff;
}

<NumberComp>
  <styled.button
    color="primary"
  >
    <StyledComponent
      color="primary"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              "
  width: 27px;
  height: 27px;
  border: 1px solid ",
              "#F62F5E",
              ";
  border-radius: 50%;
  font-size: 14px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  background-color: transparent;
  outline-color: transparent;
  outline-style: none;
  cursor: pointer;

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <button
        className="c0"
        color="primary"
      >
        0
      </button>
    </StyledComponent>
  </styled.button>
</NumberComp>
`;
